compiler/src/iree/compiler/Preprocessing/TransformExtensions/PreprocessingExtensions.cpp

```diff
@@ -303,6 +303,41 @@ IREE::transform_dialect::MatchContractionOp::matchOperation(
   return DiagnosedSilenceableFailure::success();
 }
 
+//===----------------------------------------------------------------------===//
+// MatchSizeEqualsOp
+//===----------------------------------------------------------------------===//
+
+DiagnosedSilenceableFailure
+IREE::transform_dialect::MatchSizeEqualsOp::matchOperation(
+    Operation *current, transform::TransformResults &results,
+    transform::TransformState &state) {
+  Location loc = current->getLoc();
+  ArrayRef<transform::Param> actualDimSizes =
+      state.getParams(getDimensionSizes());
+  ArrayAttr targetDimensionSizes = getExpectedValues();
+
+  if (actualDimSizes.size() != targetDimensionSizes.size()) {
+    return emitSilenceableFailure(loc)
+           << "Dimension sizes array and target sizes array have different "
+              "lengths";
+  }
+
+  for (auto [dimParam, targetValuesAttr] :
+       llvm::zip_equal(actualDimSizes, targetDimensionSizes)) {
+    int64_t currentDimSize = cast<IntegerAttr>(dimParam).getInt();
+    ArrayAttr allowedSizes = cast<ArrayAttr>(targetValuesAttr);
+    bool foundMatch = llvm::any_of(allowedSizes, [&](Attribute targetAttr) {
+      return cast<IntegerAttr>(targetAttr).getInt() == currentDimSize;
+    });
```

Comment:
How do you know you know each dim gets matched? For example, I think this would match:
expected: `[64, 64, 64]`
actual: `[64, 128, 256]`